### DISCUSSION TITLE
Configuration options menu

### DISCUSSION THREAD

**Discussion by Creomortis** [mpv-player/mpv#14566] [Ideas]
Hello, I really appreciate this application and all the effort that has been put into it, but.. Can we have an options menu built into the app, perhaps a cog on the ui to open up a box with a list of options which can be toggle with checkboxes or fields for configuration?

I honestly don't care about feeling like hackerman editing the conf file and having to look through a huge doc about commands. 

Please?

**Comment by ghost** [comment]
There are already various frontends that implement this.

https://github.com/mpv-player/mpv/wiki/Applications-using-mpv

**Reply by Creomortis** [reply]
Yeah I like as little bloat as possible so I was avoiding other projects, just wanted a quick config. I think more people would use mpv if there was a very basic simple default ui.

**Reply by N-R-K** [reply]
> I like as little bloat as possible so I was avoiding other projects

I don't understand this at all. You are avoiding frontends that add gui because it's "bloat" but then are asking to add that bloat that you're avoiding into mpv directly?

This makes absolutely zero sense. Frontends are not bloat if you need those functionalities. And adding those functionalities in mpv directly doesn't magically make it "not bloat".

**Reply by daneelg** [reply]
@N-R-K ffs, man, are you some kind of mentally impaired person or what? Since when a simple, basic "settings"/"options" menu is "bloat"?????

**Reply by N-R-K** [reply]
I'm not the one who's calling frontends "bloat" and avoiding it. Read the entire conversation properly before calling someone mentally impaired.

**Reply by daneelg** [reply]
Sorry for my language, I got carried away. Still, having a minimal menu with some essentials options is not the same as throwing an entire different frontend on top on the software! The mpv already has a graphical interface, why is it so outrageous the idea of adding a small dialog box to expose some essential parameters, instead of having to edit the .cfg file? We are in 2025, not in the 80's, it's not that difficult, ffs! It is not even something new, every damn multimedia players, from windows 3.1 onwards, had something like that. No one was asking for some new revolutionary invention.

**Reply by guidocella** [reply]
mpv's pseudo-GUI is hacked together on a subtitle rendering library where every graphical element has to be NIH'd. It is infinitely easier for frontends with real GUI libraries to add form inputs, and more robust, as they will have hardware acceleration, refresh rate not tied to the video's FPS, accessibility for free. You are welcome to try to implement a form with ASS if you want to though.

**Comment by KebabLord** [comment]
even though the project is no longer maintained, "Glow" the mpv configurator is probably what you are looking for
https://github.com/MattMcManis/Glow
![image](https://github.com/user-attachments/assets/92e4948d-2668-4645-8f6f-bceac49b80ff)

**Reply by Creomortis** [reply]
That's amazing, shame it isn't in as default.

**Reply by KebabLord** [reply]
sadly the latest version was from 4 years ago, so be ready for some renamed options and missing features.
but overall, most of the settings should work.

**Reply by Creomortis** [reply]
Thank you

**Reply by ghost** [reply]
Please don't use or link this crap. How exactly does this solve what the user is looking for? This hasn't been maintained for _four_ years, so now they have to navigate through this program AND the manual at the same time just to see new options or figure out what settings have been deprecated...

**Reply by KebabLord** [reply]
@norinoriko only this commands were removed since v2020, rest is changes on default values etc.  The script will still work.
```md
 - player: remove special 'auto' option from alang/slang/vlang (previous default)
 - vo_gpu: remove --tone-mapping-mode
 - vo_gpu: remove --scale-wblur, --scale-cutoff etc.
 - vo_gpu: remove --scaler-lut-size
 - m_option: drop support for -del for list options
 - vo_gpu_next: remove --tone-mapping-crosstalk
 - vo_gpu: remove --gamut-clipping, --gamut-warning, --tone-mapping-desaturate and --tone-mapping-desaturate-exponent
 - vulkan: remove --vulkan-disable-events
 - vo_gpu: remove --icc-contrast
 - stats.lua: remove script-opts for the main keys (i/I)
 - vo_gpu: opengl: remove --opengl-restrict
```

**Reply by llyyr** [reply]
That doesn't include changes to arguments of commands. For example, gpu-hq doesn't exist anymore. openGL PBO or PBO formats aren't options end users should ever need to set, and I'm sure there are more

**Reply by hooke007** [reply]
Once know a little bit deeper about mpv，I thouht glow is valueless.
What is "driver"? What would "PBO format" do? Why only listing these limited options?

You can understand everything in it without manual? 
You can know anything still work as it was without `interface-changes`? 
It's self-deception.

**Comment by llyyr** [comment]
The defaults should be just fine in most cases, but if you have a specific need then ctrl+f the doc and use the option that does what you need. You don't need a 100 lines long config file

**Reply by geextahslex** [reply]
My config is at 440 lines 😂 
but probably half of that is disabled

**Comment by geextahslex** [comment]
I know that for a beginner mpv can be very overwhelming and confusing but I think that no gui is what makes mpv fun to use, interact with the community (here/reddit/discord etc.) and also it is easier to develop and more stable. 
After some time you will set all your settings, scripts, [keybinds](https://github.com/mpv-player/mpv/blob/master/etc/input.conf) and make profile conditions for every scenario. The only additional software is [yt-dlp](https://github.com/yt-dlp/yt-dlp) for youtube/online stuff.
At the beginning I took some configs from other people and look what was interesting for my setup. As @llyyr mentioned most stuff should be working at default settings.
I have to agree that the manual is confusing and doesn't have the best examples. It is more like a lookup table than a real guide. 

This is only my speculation but I think that VLC is a good example on how bloated and unstable a player becomes when it has gui for everything. It was crashing on regular basis. I switched after a month to mpv and never looked back.

And you have to keep in mind that "command line" is literally in the description of the project
![command](https://github.com/user-attachments/assets/31d37821-45d1-4cf7-8a6c-26cb11c63b0e)
